// @asgui/angular/esm5/lib/sidebar/app-sidebar-form.component.js
import { __decorate, __metadata } from "tslib";
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var AppSidebarFormComponent = /** @class */ (function () {
    function AppSidebarFormComponent() {
        this.sidebarFormClass = true;
    }
    __decorate([
        HostBinding('class.sidebar-form'),
        __metadata("design:type", Object)
    ], AppSidebarFormComponent.prototype, "sidebarFormClass", void 0);
    AppSidebarFormComponent = __decorate([ __metadata("design:paramtypes", [])
    ], AppSidebarFormComponent);
AppSidebarFormComponent.ɵfac = function AppSidebarFormComponent_Factory(t) { return new (t || AppSidebarFormComponent)(); };
AppSidebarFormComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppSidebarFormComponent, selectors: [["app-sidebar-form"], ["cui-sidebar-form"]], hostVars: 2, hostBindings: function AppSidebarFormComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("sidebar-form", ctx.sidebarFormClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function AppSidebarFormComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppSidebarFormComponent, [{
        type: Component,
        args: [{
                selector: 'app-sidebar-form, cui-sidebar-form',
                template: "<ng-content></ng-content>"
            }]
    }], function () { return []; }, { sidebarFormClass: [{
            type: HostBinding,
            args: ['class.sidebar-form']
        }] }); })();
    return AppSidebarFormComponent;
}());
export { AppSidebarFormComponent };
