// @asgui/angular/esm5/lib/sidebar/app-sidebar-header.component.js
import { __decorate, __metadata } from "tslib";
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var AppSidebarHeaderComponent = /** @class */ (function () {
    function AppSidebarHeaderComponent() {
        this.sidebarHeaderClass = true;
    }
    __decorate([
        HostBinding('class.sidebar-header'),
        __metadata("design:type", Object)
    ], AppSidebarHeaderComponent.prototype, "sidebarHeaderClass", void 0);
    AppSidebarHeaderComponent = __decorate([ __metadata("design:paramtypes", [])
    ], AppSidebarHeaderComponent);
AppSidebarHeaderComponent.ɵfac = function AppSidebarHeaderComponent_Factory(t) { return new (t || AppSidebarHeaderComponent)(); };
AppSidebarHeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppSidebarHeaderComponent, selectors: [["app-sidebar-header"], ["cui-sidebar-header"]], hostVars: 2, hostBindings: function AppSidebarHeaderComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("sidebar-header", ctx.sidebarHeaderClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function AppSidebarHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppSidebarHeaderComponent, [{
        type: Component,
        args: [{
                selector: 'app-sidebar-header, cui-sidebar-header',
                template: "<ng-content></ng-content>"
            }]
    }], function () { return []; }, { sidebarHeaderClass: [{
            type: HostBinding,
            args: ['class.sidebar-header']
        }] }); })();
    return AppSidebarHeaderComponent;
}());
export { AppSidebarHeaderComponent };
