// @coreui/angular/esm2015/lib/sidebar/app-sidebar-footer.component.js
import { __decorate, __metadata } from "tslib";
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
let AppSidebarFooterComponent = class AppSidebarFooterComponent {
    constructor() {
        this.sidebarFooterClass = true;
    }
};
AppSidebarFooterComponent.ɵfac = function AppSidebarFooterComponent_Factory(t) { return new (t || AppSidebarFooterComponent)(); };
AppSidebarFooterComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppSidebarFooterComponent, selectors: [["app-sidebar-footer"], ["cui-sidebar-footer"]], hostVars: 2, hostBindings: function AppSidebarFooterComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("sidebar-footer", ctx.sidebarFooterClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function AppSidebarFooterComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
__decorate([
    HostBinding('class.sidebar-footer'),
    __metadata("design:type", Object)
], AppSidebarFooterComponent.prototype, "sidebarFooterClass", void 0);
AppSidebarFooterComponent = __decorate([ __metadata("design:paramtypes", [])
], AppSidebarFooterComponent);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppSidebarFooterComponent, [{
        type: Component,
        args: [{
                selector: 'app-sidebar-footer, cui-sidebar-footer',
                template: `<ng-content></ng-content>`
            }]
    }], function () { return []; }, { sidebarFooterClass: [{
            type: HostBinding,
            args: ['class.sidebar-footer']
        }] }); })();
export { AppSidebarFooterComponent };
