// @asgui/angular/esm5/lib/footer/app-footer.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AppFooterComponent } from './app-footer.component';
import * as ɵngcc0 from '@angular/core';
var AppFooterModule = /** @class */ (function () {
    function AppFooterModule() {
    }
AppFooterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppFooterModule });
AppFooterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppFooterModule_Factory(t) { return new (t || AppFooterModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppFooterModule, { declarations: function () { return [AppFooterComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [AppFooterComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppFooterModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [AppFooterComponent],
                declarations: [AppFooterComponent]
            }]
    }], function () { return []; }, null); })();
    return AppFooterModule;
}());
export { AppFooterModule };
