// @asgui/angular/esm5/lib/sidebar/app-sidebar-nav/app-sidebar-nav-divider.component.js
import { __decorate, __metadata } from "tslib";
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var AppSidebarNavDividerComponent = /** @class */ (function () {
    function AppSidebarNavDividerComponent() {
    }
    AppSidebarNavDividerComponent.prototype.ngOnInit = function () { };
    __decorate([
        Input(),
        __metadata("design:type", Object)
    ], AppSidebarNavDividerComponent.prototype, "item", void 0);
    AppSidebarNavDividerComponent = __decorate([ __metadata("design:paramtypes", [])
    ], AppSidebarNavDividerComponent);
AppSidebarNavDividerComponent.ɵfac = function AppSidebarNavDividerComponent_Factory(t) { return new (t || AppSidebarNavDividerComponent)(); };
AppSidebarNavDividerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AppSidebarNavDividerComponent, selectors: [["app-sidebar-nav-divider"], ["cui-sidebar-nav-divider"]], inputs: { item: "item" }, decls: 0, vars: 0, template: function AppSidebarNavDividerComponent_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppSidebarNavDividerComponent, [{
        type: Component,
        args: [{
                selector: 'app-sidebar-nav-divider, cui-sidebar-nav-divider',
                template: ""
            }]
    }], function () { return []; }, { item: [{
            type: Input
        }] }); })();
    return AppSidebarNavDividerComponent;
}());
export { AppSidebarNavDividerComponent };
