// @coreui/angular/esm5/lib/aside/app-aside.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LayoutModule } from './../shared/layout/layout.module';
import { AppAsideComponent } from './app-aside.component';
import * as ɵngcc0 from '@angular/core';
var AppAsideModule = /** @class */ (function () {
    function AppAsideModule() {
    }
AppAsideModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppAsideModule });
AppAsideModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppAsideModule_Factory(t) { return new (t || AppAsideModule)(); }, imports: [[
            CommonModule,
            LayoutModule
        ],
        LayoutModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppAsideModule, { declarations: function () { return [AppAsideComponent]; }, imports: function () { return [CommonModule,
        LayoutModule]; }, exports: function () { return [AppAsideComponent,
        LayoutModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppAsideModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    LayoutModule
                ],
                exports: [
                    AppAsideComponent,
                    LayoutModule
                ],
                declarations: [
                    AppAsideComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return AppAsideModule;
}());
export { AppAsideModule };
