// @asgui/angular/esm5/lib/header/app-header.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { LayoutModule } from '../shared/layout/layout.module';
import { AppHeaderComponent } from './app-header.component';
import * as ɵngcc0 from '@angular/core';
var AppHeaderModule = /** @class */ (function () {
    function AppHeaderModule() {
    }
AppHeaderModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AppHeaderModule });
AppHeaderModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AppHeaderModule_Factory(t) { return new (t || AppHeaderModule)(); }, imports: [[
            CommonModule,
            RouterModule,
            LayoutModule
        ],
        LayoutModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AppHeaderModule, { declarations: function () { return [AppHeaderComponent]; }, imports: function () { return [CommonModule,
        RouterModule,
        LayoutModule]; }, exports: function () { return [AppHeaderComponent,
        LayoutModule]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AppHeaderModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    RouterModule,
                    LayoutModule
                ],
                exports: [
                    AppHeaderComponent,
                    LayoutModule
                ],
                declarations: [
                    AppHeaderComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return AppHeaderModule;
}());
export { AppHeaderModule };
